src/components/back-to-top.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useEffect, useState } from "react";
+import { FaCaretUp } from "react-icons/fa";
+
+const BackToTop: React.FC<{}> = () => {
+  const [visible, setVisible] = useState(false);
+
+  const listener: EventListener = () => {
+    if (window.scrollY > 0) {
+      setVisible(true);
+    } else {
+      setVisible(false);
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener("scroll", listener);
+    return (): void => {
+      window.removeEventListener("scroll", listener);
+    };
+  });
+
+  const scrollToTop = (): void => {
+    window.scrollTo({
+      top: 0,
+      left: 0,
+      behavior: "smooth",
+    });
+  };
+
+  return (
+    <div
```

Comment:
_nitpick_
use `<a> or <button>`  to follow html semantics. I'll let you make the call to edit it or not
Link: https://stackoverflow.com/a/2777280

Reply:
Good point. For accessibility purpose too. Will make the change.